art/attacks/boundary.py

```diff
@@ -145,60 +162,59 @@ def _attack(self, initial_sample, original_sample, target, initial_delta, initia
         :return: an adversarial example.
         :rtype: `np.ndarray`
         """
-        def compare(object1, object2):
-            return object1 == object2 if self.targeted else object1 != object2
-
         # Get initialization for some variables
         x_adv = initial_sample
-        delta = initial_delta
-        epsilon = initial_epsilon
+        self.curr_delta = initial_delta
+        self.curr_epsilon = initial_epsilon
+        clip_min, clip_max = self.classifier.clip_values
 
         # Main loop to wander around the boundary
-        for _ in range(self.max_iter):
+        for m in range(self.max_iter):
```

Comment:
Do we need to name the variable `m`? It seems it is not used inside of the loop.